eth/catalyst/api.go

```diff
@@ -163,15 +197,56 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update beacon.ForkchoiceStateV1, pa
 		data, err := api.assembleBlock(update.HeadBlockHash, payloadAttributes)
 		if err != nil {
 			log.Error("Failed to create sealing payload", "err", err)
-			return beacon.ForkChoiceResponse{Status: beacon.VALID}, err // Valid as setHead was accepted
+			return beacon.STATUS_INVALID, err // Valid as setHead was accepted
 		}
 		id := computePayloadId(update.HeadBlockHash, payloadAttributes)
 		api.localBlocks.put(id, data)
 
 		log.Info("Created payload for sealing", "id", id, "elapsed", time.Since(start))
-		return beacon.ForkChoiceResponse{Status: beacon.VALID, PayloadID: &id}, nil
+		return api.validFCU(&id), nil
 	}
-	return beacon.ForkChoiceResponse{Status: beacon.VALID}, nil
+	return api.validFCU(nil), nil
+}
+
+func (api *ConsensusAPI) validFCU(id *beacon.PayloadID) beacon.ForkChoiceResponse {
+	currentHash := api.eth.BlockChain().CurrentBlock().Hash()
+	return beacon.ForkChoiceResponse{
+		PayloadStatus: beacon.PayloadStatusV1{Status: beacon.VALID, LatestValidHash: &currentHash},
+		PayloadID:     id,
+	}
+}
+
+func (api *ConsensusAPI) ExchangeTransitionConfigurationV1(config beacon.TransitionConfigurationV1) (*beacon.TransitionConfigurationV1, error) {
+	if config.TerminalTotalDifficulty == nil {
+		return nil, errors.New("invalid terminal total difficulty")
+	}
+	ttd := api.eth.BlockChain().Config().TerminalTotalDifficulty
+	if ttd.Cmp(config.TerminalTotalDifficulty.ToInt()) != 0 {
+		return nil, fmt.Errorf("invalid ttd: EL %v CL %v", ttd, config.TerminalTotalDifficulty)
+	}
+	terminalHeader := api.eth.BlockChain().CurrentTerminalHeader()
+	if terminalHeader != nil {
+
+		if config.TerminalBlockNumber != 0 && uint64(config.TerminalBlockNumber) != terminalHeader.Number.Uint64() {
+			return nil, fmt.Errorf("invalid terminal block number, got %v want %v", config.TerminalBlockNumber, terminalHeader.Number)
+		}
+
+		if config.TerminalBlockHash != (common.Hash{}) && config.TerminalBlockHash != terminalHeader.Hash() {
+			return nil, fmt.Errorf("invalid terminal block hash, got %v want %v", config.TerminalBlockHash, terminalHeader.Hash())
+		}
+		// TODO (MariusVanDerWijden): Here we used to return the correct terminal hash + number
+		// but according to the spec we should return the hash + number set by the user.
+		// Since we have no way of setting hash + number yet, just return TTD.
+		return &beacon.TransitionConfigurationV1{TerminalTotalDifficulty: (*hexutil.Big)(ttd)}, nil
+	}
+	if config.TerminalBlockNumber != 0 {
```

Comment:
I'd guess that 0 would be a valid terminal block number if a new network is started that's already merged from genesis